config/initializers/register_mail_interceptors.rb

```diff
@@ -29,8 +29,6 @@
 # Register interceptors defined in app/mailers/user_mailer.rb
 # Do this here, so they aren't registered multiple times due to reloading in development mode.
 
-Rails.application.config.action_mailer.interceptors = [
-  "DefaultHeadersInterceptor",
-  # following needs to be the last interceptor
-  "DoNotSendMailsWithoutReceiverInterceptor"
-]
+ApplicationMailer.register_interceptor Interceptors::DefaultHeaders
+# following needs to be the last interceptor
+ApplicationMailer.register_interceptor Interceptors::DoNotSendMailsWithoutRecipient
```

Comment:
The [guide on interceptors](https://guides.rubyonrails.org/action_mailer_basics.html#intercepting-emails) suggests to wrap code like this in a configure block: 

```
Rails.application.configure do
  config.action_mailer.interceptors = %w[Interceptors::DefaultHeaders Interceptors::DoNotSendMailsWithoutRecipient]
end
```

Doing this now should make it easier for us to migrate to rails 7

Reply:
This is the exact line that breaks the change, as the ActionMailer railtie is executed before this initializer gets run, causing the interceptors to be empty. It is not called anymore afterwards, regardless of whether it is wrapped in a configure block or not.

I only traced it back to the health checks initializer which has an SMTP health check accessing ActionMailer. If we can fix that, it might be able to resolve. For 12.1, this doesn't appear feasible however.

Reply:
Ok, then let's leave it at that and fix the problem during the rails 7 migration.